src/Dialect/ONNX/ONNXOps/Canonicalize.td

```diff
@@ -334,8 +334,9 @@ def FuseAddConvNullBiasPattern: Pat<
   [(HasShapeAndRank:$res),
    (HasNoneType $b),
    (AttributeIsNotNull:$denseAttr),
+   (RankXMinusRankYIs<1> $res, $y),
```

Comment:
I think this happens that `$res` has rank of 4 in this case. But if `$res` has rank of 1 and `$y` is a scalar (rank 0), `RankXMinusRankYIs<1> $res, $y` is still true.

A better way would be check that `$y` is not  scalar by adding `HasRankGT<0> $y`, which says `$y` must have rank greater than 0.